File: backend/CinemaReservation/CinemaReservation.Application/DTOs/SeatAvailabilityDto.cs
// Application/DTOs/SeatAvailabilityDto.cs

public class SeatAvailabilityDto
{
    public int BillboardId { get; set; }
    public string RoomName { get; set; }
    public int TotalSeats { get; set; }
    public int OccupiedSeats { get; set; }
    public int AvailableSeats { get; set; }
    public List<SeatDto> SeatDetails { get; set; } = new();
}

public class SeatDto
{
    public int Id { get; set; }
    public short Number { get; set; }
    public short Row { get; set; }
    public bool IsOccupied { get; set; }
}

File: backend/client/src/types/SeatAvailabilityDto.ts
// src/types/SeatAvailabilityDto.ts

export interface SeatAvailabilityDto {
    billboardId: number;
    roomName: string;
    totalSeats: number;
    occupiedSeats: number;
    availableSeats: number;
    seatDetails: SeatDto[];
  }
  
  export interface SeatDto {
    id: number;
    number: number;
    row: number;
    isOccupied: boolean;
  }
  